### DISCUSSION TITLE
Separate work use and personal use

### DISCUSSION THREAD

**Discussion by Jiayou-Chao** [community/community#144151] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

According to GitHub's Terms of Service, I should use only one account. However, this creates a security issue with SSH keys.

When I set up SSH keys on my work computer, I realize these keys will have full access to all my repositories. This means if someone gains access to my work computer's SSH key, they could potentially access my personal repositories as well.

Questions:
1. How do you handle this situation?
2. Is there a way to limit SSH key access to specific repositories?
3. What's the recommended approach for maintaining security between work and personal projects while complying with GitHub's single-account policy?

**Comment by Cardosofiles** [comment]
I am using the authentication application: https://play.google.com/store/apps/details?id=com.azure.authenticator&referrer=%20adjust_reftag%3Dc6f1p4ErudH2C%26utm_source%3DLanding%2BPage%2BOrganic%2B-%2Bapp%2Bstore%2Bbadges%26utm_campaign%3Dappstore_android to perform the validations.

You register your GitHub account, and when you request passwords, you use the application.

There is the option of the GitHub application itself, where it generates the code for user validation.

**Reply by Jiayou-Chao** [reply]
I hope there is a more convenient solution. I don't want to have to search for my phone and authorize each fetch/pull/push request.

**Reply by Cardosofiles** [reply]
I understand, it can be a real nuisance.

**Comment by dalvarez2596** [comment]
Hi @Jiayou-Chao I found this document on GitHub about switching between accounts that might be useful. I also checked the terms, and depending on the situation, it seems that you can only have one free account associated with a single email. For example, in my previous job, I had a GitHub account associated with my work email and a separate personal account. So, it seems like you can't have more than one free account linked to the same email. Otherwise, you wouldn't find any documents in the GitHub docs about switching between accounts, like the ones below:
[Switching between accounts](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/switching-between-accounts)
[Set up your SSH keys](https://gist.github.com/rahularity/86da20fe3858e6b311de068201d279e3)

Additionally, some companies provide paid accounts for employees, so in that case, it might work.

**Reply by Jiayou-Chao** [reply]
Thank you. It seems to be helpful. I will take a closer look at it.

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @Jiayou-Chao!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the General category. I've gone ahead and moved it for you. Good luck!
